### ISSUE TITLE
Model predictions on data with a different `sample_rate`

### ISSUE THREAD

**Issue by charleygros** [kitzeslab/opensoundscape#687]
Hi there,

Question for you:
I have been training a model on data with a sampling rate of 24000.
Now doing some predictions on data with a sampling rate of 32000.

I am guessing that it would be good to the same sampling rate than the training data?!

Does the model preprocessor will do the resampling to 24000 by default?
Or should I add a resample action in the preprocessor at testing time to enforce this? If so, could you please guide me.

Many thanks in advance.
Charley

**Comment by sammlapp** [comment]
Hi @charleygros, the short answer is that the model will work fine on your audio data without any modifications.

In the default preprocessing pipeline of the CNN class, spectrograms are bandpassed to 0-11025 Hz. Therefore, regardless of whether the input audio has a sample rate of 24000 or 32000 Hz, the spectrograms generated and passed to the CNN will span the frequency range 0-11025 Hz. 

There is one slight difference that will occur, however. The Spectrogram window_samples parameter (default of 512 samples) will mean something slightly different for 24 kHz vs 32 kHz audio files (ie each window will represent a slightly different amount of time). The effect of this on the spectrograms produced is likely to be negligible. However, if you want to resample audio to 24 kHz during preprocessing to ensure equivalency in this respect, you could manipulate the preprocessor as follows:
```
# let's assume your object of the CNN class is called `model`
# change the sample_rate of the load_audio Action in the preprocessor's pipeline from the default value of None to 24000
# (the default value of None means that audio is not resampled during preprocessing) 
model.preprocessor.pipeline.load_audio.set(sample_rate=24000)
```
The disadvantage to this is that resampling audio is slow and will slow down preprocessing significantly.

**Comment by charleygros** [comment]
Many thanks for your prompt and very useful answer @sammlapp 
Really appreciated

I did a quick test on my dataset using the suggested command: the predictions were almost identical, not exactly, but very similar. And much slower, as you previously said.

We can close the issue, I think, solved as far as I'm concerned.

(I'm taking the opportunity to share this with you and your team: it has been a real pleasure to play around with your library. Only one week in, but I made the decision to change all my processing pipeline to use your tools instead of others. Much appreciation. Keep it up!)
